File: Task1.py
import math

# Information for each product
productA = int(input("Enter the quantity of product A: "))
wrap_productA = input("Is product A wrapped(y/n): ")
productB = int(input("Enter the quantity of product B: "))
wrap_productB = input("Is product B wrapped(y/n): ")
productC = int(input("Enter the quantity of product C: "))
wrap_productC = input("Is product C wrapped(y/n): ")

# Calculating subtotals for each product
subtotal_productA = 20 * productA
subtotal_productB = 40 * productB
subtotal_productC = 50 * productC

# Calculating total estimate without discounts and other charges
subtotal = subtotal_productA + subtotal_productB + subtotal_productC


# Discounts
flat_10_discount = 10 if subtotal > 200 else 0
bulk_5_discount = (5 * subtotal) / 100 if max(productA, productB, productC) > 10 else 0
bulk_10_discount = (10 * subtotal) / 100 if productA + productB + productC > 20 else 0
tiered_50_discount = 0

if productA > 15:
    tiered_50_discount += 0.5 * subtotal_productA
if productB > 15:
    tiered_50_discount += 0.5 * subtotal_productB
if productC > 15:
    tiered_50_discount += 0.5 * subtotal_productC

# Determining the maximum discount
max_discount = max(flat_10_discount, bulk_5_discount, bulk_10_discount, tiered_50_discount)

# Examinig the discount category
if max_discount == flat_10_discount:
    discount_category = "flat_10_discount"
elif max_discount == bulk_5_discount:
    discount_category = "bulk_5_discount"
elif max_discount == bulk_10_discount:
    discount_category = "bulk_10_discount"
else:
    discount_category = "tiered_50_discount"



# Calculation of wrapping charges
wrap_fee = sum([productA if wrap_productA == "y" else 0, productB if wrap_productB == "y" else 0,
                productC if wrap_productC == "y" else 0])

# Delivery charges
shipping_fee = sum([math.ceil(productA / 10) * 5, math.ceil(productB / 10) * 5,
                    math.ceil(productC / 10) * 5])

# Calculation of total fee
total_fee = subtotal + wrap_fee + shipping_fee - max_discount


print("Product A, Quantity:", productA)
print("Total amount of Product A :",subtotal_productA)
print("Product B, Quantity :", productB)
print("Total amount of Product B :",subtotal_productB)
print("Product C, Quantity :", productC)
print("Total amount of Product C :",subtotal_productC)
print("Sub-Total :", subtotal)

# discount information
print("Discount applied:", discount_category)
print("Discount amount:", max_discount)

# Display wrapping and delivery charges
print("Shipping fee:", shipping_fee)
print("Gift Wrap fee:", wrap_fee)
print("Total fee:", total_fee)

File: Task_1.js
const readlineSync = require('readline-sync');

// Information for each product
const productA = parseInt(readlineSync.question("Enter the quantity of product A: "));
const wrap_productA = readlineSync.question("Is product A wrapped(y/n): ");
const productB = parseInt(readlineSync.question("Enter the quantity of product B: "));
const wrap_productB = readlineSync.question("Is product B wrapped(y/n): ");
const productC = parseInt(readlineSync.question("Enter the quantity of product C: "));
const wrap_productC = readlineSync.question("Is product C wrapped(y/n): ");

// Calculating subtotals for each product
const subtotal_productA = 20 * productA;
const subtotal_productB = 40 * productB;
const subtotal_productC = 50 * productC;

// Calculating total estimate without discounts and other charges
const subtotal = subtotal_productA + subtotal_productB + subtotal_productC;

// Discounts
const flat_10_discount = subtotal > 200 ? 10 : 0;
const bulk_5_discount = (5 * subtotal) / 100 > Math.max(productA, productB, productC) ? (5 * subtotal) / 100 : 0;
const bulk_10_discount = (10 * subtotal) / 100 > productA + productB + productC ? (10 * subtotal) / 100 : 0;
let tiered_50_discount = 0;

if (productA > 15) {
    tiered_50_discount += 0.5 * subtotal_productA;
}
if (productB > 15) {
    tiered_50_discount += 0.5 * subtotal_productB;
}
if (productC > 15) {
    tiered_50_discount += 0.5 * subtotal_productC;
}

// Determining the maximum discount
const max_discount = Math.max(flat_10_discount, bulk_5_discount, bulk_10_discount, tiered_50_discount);

// Examining the discount category
let discount_category;

if (max_discount === flat_10_discount) {
    discount_category = "flat_10_discount";
} else if (max_discount === bulk_5_discount) {
    discount_category = "bulk_5_discount";
} else if (max_discount === bulk_10_discount) {
    discount_category = "bulk_10_discount";
} else {
    discount_category = "tiered_50_discount";
}

// Calculation of wrapping charges
const wrap_fee = (wrap_productA === "y" ? productA : 0) + (wrap_productB === "y" ? productB : 0) + (wrap_productC === "y" ? productC : 0);

// Delivery charges
const shipping_fee = Math.ceil(productA / 10) * 5 + Math.ceil(productB / 10) * 5 + Math.ceil(productC / 10) * 5;

// Calculation of total fee
const total_fee = subtotal + wrap_fee + shipping_fee - max_discount;

// Displaying the results
console.log(`Product A, Quantity: ${productA}`);
console.log(`Total amount of Product A: ${subtotal_productA}`);
console.log(`Product B, Quantity: ${productB}`);
console.log(`Total amount of Product B: ${subtotal_productB}`);
console.log(`Product C, Quantity: ${productC}`);
console.log(`Total amount of Product C: ${subtotal_productC}`);
console.log(`Sub-Total: ${subtotal}`);


console.log(`Discount applied: ${discount_category}`);
console.log(`Discount amount: ${max_discount}`);

// Displaying wrapping and delivery charges
console.log(`Shipping fee: ${shipping_fee}`);
console.log(`Gift Wrap fee: ${wrap_fee}`);
console.log(`Total fee: ${total_fee}`);
